content/news/XXXX-XX-XX-dvs-internals-pt1.md

```diff
@@ -0,0 +1,115 @@
+
+But how does the computer know how many cycles there are in a second?
+We can't actually wait for a second, because the delay (also called latency) would be very noticeable and make the system unsuitable for scratching.
+
+## Zero Crossings and Digital Samples
+
+A simple low-latency approach to detect the pitch is using so-called "zero crossings".
+Whenever the wave goes from the positive part to the negative part or vice-versa, it crosses the black line.
+That lines denotes zero, which is why this is called a "zero crossing".
+
+It's important to know that in contrast to analog audio like vinyl or music cassettes, the computer doesn't actually work with the continuous wave that we see in the screenshot.
+Instead, [digital signals](https://en.wikipedia.org/wiki/Digital_signal_(signal_processing)) works with measurements (called "samples") that indicate the value of the wave position at specific points in time.
+How often these measurements are taken is determined by the sample rate (or sampling rate).
+For example, Audio CDs use a sample rate of 44100 Hz (i.e. 44100 measurements per second, or 1 measurement every ~22.68 microseconds).
+
+To illustrate this, here are the individual samples in the timecode signal signified by little dots:
+
+![Individual samples in timecode signal]({static}/images/news/timecode-signal-samples.png)
+
+To detect zero crossings reliably, we cannot assume that such a sample measurement is happening at the exact time that the signal is at zero position.
+It might be the case the current sample is *after* the zero position and the previous sample was *before* the zero position.
+Hence, we also compare the previous sample value with the current sample value.
+If the current value is positive and the previous value was negative, or if the previous value was negative and the current value is positive, we crossed zero.
+
+
+## Refining Pitch Detection using Zero Crossings
+
+Now that we know how to detect zero crossings, we can use them for determining the pitch.
+
+If we look at the timecode signal, we can see that each cycle of the timecode signal has two zero crossings:
+At the start of the cycle, when the positive part of the wave start and in the middle of the cycle when the negative part of the wave starts.
+
+So the equivalent of checking for 1000 cycles per second is to check if there are 2000 zero crossings per second.
+Let's say the audio interface uses a sample rate of 44100 Hz, then there should be a be a zero crossing every 441000 / 2000 = 22.05 samples.
+
+We can now detect the pitch by comparing the expected number of samples between zero crossings with the actual number of samples between them.
+However, we need to use multiple sample distances and calculate the average to make this calculation more accurate, because in reality there are no fractions of samples, only full samples.
+
+
```

Comment:
This algorithm is very simple and works at low latencies. It would not work for a music signal which has lots of different frequencies added together. The Serato timecode signal only has a single frequency of 1 kHz, so it works for this case.